### ISSUE TITLE
False Positive: MD063 corrects e.g. `## May 5th` to `## May 5Th`

### ISSUE THREAD

**Issue by athyfr** [rvben/rumdl#608]
### rumdl version

0.1.42

### Installation method

Other

### Bug description

When `MD063` corrects headers to title-case, it'll correct numerals such as `6th` to capitalize the ending `6Th`, which is clearly incorrect.

### Steps to reproduce

1. Create a `test.md` file with the following content
   ```md
   # 1st Test Markdown File For This Issue
   
   Also the only test file
   ```
2. In the same directory, create a `.rumdl.toml` file with `MD063` enabled through `extend-enable`.
3. Run `rumdl check`
4. Observe the incorrect lint.

### Expected behavior

`May 3rd` as a header should not produce any linter errors.

`May 3Rd` should correct to `May 3rd`.

### Actual behavior

`May 3rd` corrects to `May 3Rd`.

`May 3Rd` is interpreted as correct.

### Configuration

```toml
[global]
extend-enable = ["MD063"]
```

### Operating system

Linux

**Comment by rvben** [comment]
Fixed in v0.1.88. Thanks for the report, @athyfr!

Added an `is_numeric_ordinal()` check so suffixes like `st`, `nd`, `rd`, and `th` following a digit are always kept lowercase during title-case conversion, matching the expected behaviour for ordinals like "1st", "2nd", "3rd", "4th".
